src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -182,10 +182,13 @@ public static bool TryExtractGraphQLFieldModelName(IDirectiveCollection fieldDir
             {
                 if (dir.Name.Value == ModelDirectiveType.DirectiveName)
                 {
-                    dir.ToObject<ModelDirectiveType>();
-                    modelName = dir.GetArgument<string>(ModelDirectiveType.ModelNameArgument).ToString();
+                    ModelDirectiveType modelDirectiveType = dir.ToObject<ModelDirectiveType>();
 
-                    return modelName is not null;
+                    if(modelDirectiveType.Name.HasValue)
```

Comment:
Could you please point to which change specifically caused this regression in 0.8 for better understanding? We have release branches here: `release/0.8`, `release/0.7`

Reply:
We introduced this change in release/0.8
https://github.com/Azure/data-api-builder/blob/99f1910afeec044813d69d8badfd901e99441e68/src/Core/Resolvers/CosmosQueryStructure.cs#L120 

In earlier release we were not using this 
https://github.com/Azure/data-api-builder/blob/61de247acf65280d93783072226f695536591ffb/src/Service/Resolvers/CosmosQueryStructure.cs#L123C1-L124C1

Reply:
Thank you @neeraj-sharma2592 for identifying this!